src/modules/modules.py

```diff
@@ -270,9 +273,39 @@ def forward(self, sent, task):
         word_mask = util.get_text_field_mask(sent).float()
         return word_embs, word_mask  # need to get # nonzero elts
 
+class PRPNPhraseLayer(Model):
+    ''' PRPN phrase layer '''
```

Comment:
I was hoping for something that would actually say what PRPN (and ON) are, since most people who use this file will have never seen LTL.